src/file.cpp

```diff
@@ -1153,23 +1179,26 @@ namespace libtorrent
 		m_inode = 0;
 		// the path passed to FindFirstFile() must be
 		// a pattern
-		std::string f = convert_separators(path);
-		if (!f.empty() && f[f.size()-1] != '\\') f += "\\*";
-		else f += "*";
+		std::string pattern_path{ path };
+		if (!pattern_path.empty() && pattern_path.back() != '\\')
```

Comment:
I'm thinking of ways to make this patch a bit less verbose. what do you think about this?
``patter_path += (pattern_path.empty() && pattern_path.back() != '\\') ? "\\*" : "*";``
perhaps it would even make sense to have a utility function to check if a path ends with a "/" or a "\".